### PR TITLE
sym-node: support `print` for debugging

### PR DESCRIPTION
**Pull request by iblis17**
Ref: https://github.com/apache/incubator-mxnet/blob/470b56437290b33fef51b067c96fdce08cefa584/include/mxnet/c_api.h#L933

example: 
```julia
julia> print(mx.dot(mx.Variable(:x), mx.Variable(:y)))
Symbol Outputs:
        output[0]=dot4(0)
Variable:y
Variable:x
--------------------
Op:dot, Name=dot4
Inputs:
        arg[0]=y(0) version=0
        arg[1]=x(0) version=0
```

### CODE REVIEW

File: src/symbolic-node.jl [hunk 1]
```diff
@@ -220,6 +220,51 @@ function get_name(self :: mx.SymbolicNode)
     return Symbol(unsafe_string(name[]))
 end
 
+import Base: print
+
+function print(io :: IO, sym :: SymbolicNode)
+  out = Ref{mx.char_p}(C_NULL)
+  @mx.mxcall(:MXSymbolPrint, (mx.MX_SymbolHandle, Ref{mx.char_p}), sym.handle, out)
```
Comment (vchuravy) [inline src/symbolic-node.jl]: Who is responsible for deallocating the string?

Comment (iblis17) [reply src/symbolic-node.jl]: https://github.com/apache/incubator-mxnet/blob/470b56437290b33fef51b067c96fdce08cefa584/include/mxnet/c_api.h#L933

```
MXNET_DLL int MXSymbolPrint(SymbolHandle symbol, const char **out_str);
```

(I'm not skilled at C) 
Do I need to free the content of `const char**`?

**Comment by iblis17** [comment]
rebased for macOS build

**Comment by iblis17** [comment]
found that there is a debug api for executor as well, maybe I should rename it as well?
https://github.com/dmlc/MXNet.jl/blob/2c5120d139fdd259f2d1b161ec94eea323d0fc20/src/executor.jl#L205-L221

File: src/symbolic-node.jl (hunk 1 above)
Comment (vchuravy) [reply src/symbolic-node.jl]: I think technically we are taking  ownership of this ptr so yeah we need to free it after we are done with it, (probably elsewhere as well.)

**Comment by vchuravy** [comment]
The problem with the executor debug string is that is very large and not necessarily a useful representation. in #88 I convert the network to a digraph to get a more visual representation

**Comment by iblis17** [comment]
So... you want to remove debug_str api of executor?
How about adding #88 via specifying MIME of show?

**Comment by vchuravy** [comment]
I don't want to remove it, but it is not a useful output for just `print`. But yeah if we can "render" it nicely depending on if we are outputting text or are running in a Jupyter session that would be awesome.

**Comment by iblis17** [comment]
I see.
Let's back to symbolic node's debugging print. Is it okay to have a `print` API for it? I prefer to keep the consistency, let user using the same API.

**Comment by vchuravy** [comment]
Sure!

**Comment by vchuravy** [comment]
Can you rebase this for CI?

**Comment by iblis17** [comment]
rebased

**Pull request merged**